### DISCUSSION TITLE
BLE API to amazfit watch

### DISCUSSION THREAD

**Discussion by Sebo-the-tramp** [zepp-health/zeppos-forum#366] [Developer Feedback]
Good evening,

I hope this is the right place to ask. I was writing my iOS application and wanted to connect and download data from my amazfit smartwatch. I could not find any examples of such API calls. Could you please direct me in the right direction? I am looking for something like [BLE API](https://gopro.github.io/OpenGoPro/ble/index.html#). 
I am writing my application in Flutter. 
I searched the previous discussion where it stated that in Zepp OS 3 this should be somewhat available.

Thank you so much I appreciate your time in advance, 

Sincerely,
Sebastian

**Comment by silver-zepp** [comment]
#### Hey @Sebo-the-tramp. The functionality to extract history data isn't available, however, you can collect data for yourself and send it either to your 
- [BLE Slave device](https://github.com/silver-zepp/zeppos-easy-ble) (pay attention to the ESP32 example)
#### or a web server with one of these examples:
- [MongoDB](https://github.com/orgs/zepp-health/discussions/276)
- [ngrok](https://github.com/zepp-health/autogui/tree/master/other-autogui-examples/webserver-data-extraction)

**Comment by app4g** [comment]
hi @silver-zepp may I know if this is still true currently? Zepp does not send historical data to any Zepp servers which developer can then query and obtain user's data. Both wellness and workouts?

**Reply by jpgfigueired0** [reply]
I would also like to know if there was an update for this.
